### ISSUE TITLE
Feat: Set External Auth for Sonarr v4

### ISSUE THREAD

**Issue by JigSawFr** [saltyorg/Saltbox#117]
For Sonarr role:

IF
1. `sonarr_docker_image_tag` is set to `v4`
2. `sonarr_traefik_sso_middleware` is set to Authelia default SSO

Execute a subtask to set AuthMethod to
`<AuthenticationMethod>External</AuthenticationMethod>`

Cf. https://wiki.servarr.com/sonarr/faq-v4#can-i-disable-forced-authentication

**Comment by JigSawFr** [comment]
Need also to set `<UrlBase></UrlBase>`

**Comment by owine** [comment]
Can potentially apply similar logic to other arr apps

**Comment by JigSawFr** [comment]
> Can potentially apply similar logic to other arr apps
> 
> An alternative (this is what I currently use) would be to provision a middleware to inject basic auth credentials and apply to the arr containers

Yep, was doing it until now, prefer to use it on my side in last resort.
As they offer external choice, it is better if you use health check for e.g. (And avoid multiple BA failures in logs :>)

But yeah there is other's vars that we can automate :)

**Comment by saltydk** [comment]
Do we really need to touch existing setups?

**Comment by JigSawFr** [comment]
> Do we really need to touch existing setups?

I see it as an improvement only

**Comment by saltydk** [comment]
Well, I'd be against merging checks against user inventory edits. If this needs to be a thing then wait until v4 is the standard version.

**Comment by owine** [comment]
This is coming to Prowlarr stable soon fyi

**Comment by JigSawFr** [comment]
> This is coming to Prowlarr stable soon fyi

Thanks for info !

**Comment by saltydk** [comment]
> Need also to set `<UrlBase></UrlBase>`

That is the default so not sure why this is mentioned.

**Comment by saltydk** [comment]
So maybe something like 
```yaml
- name: Lookup AuthenticationMethod value
  community.general.xml:
    path: "/opt/{{ prowlarr_name }}/config.xml"
    xpath: "/Config/AuthenticationMethod"
    content: "text"
  register: xmlresp

- name: Toggle AuthenticationMethod to External
  block:
    - name: Remove the 'subjective' attribute of the 'rating' element
      community.general.xml:
        path: "/opt/{{ prowlarr_name }}/config.xml"
        xpath: "/Config/AuthenticationMethod"
        value: "External"

    - name: Restart Docker container
      ansible.builtin.include_tasks: "{{ resources_tasks_path }}/docker/restart_docker_container.yml"

  when: ((xmlresp.matches[0].AuthenticationMethod is defined) and (xmlresp.matches[0].AuthenticationMethod != 'External'))```

**Comment by JigSawFr** [comment]
@saltydk yes, maybe we can add also another condition to check if  sso middleware is defined to authelia, so we can keep initial behavior in case of ?

**Comment by saltydk** [comment]
I'd just have that in the include of the subtask

**Comment by JigSawFr** [comment]
> > Need also to set `<UrlBase></UrlBase>`
> 
> 
> 
> That is the default so not sure why this is mentioned.

Not sure how it's used in these programs, but in case of to avoid future problems ? At the moment it's working as expected without editing it.

**Comment by saltydk** [comment]
They won't set a UrlBase for you so don't worry about it.

**Comment by saltydk** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base

**Comment by saltydk** [comment]
https://github.com/saltyorg/Saltbox/commit/a8571593f49a2d442e96da6a7fed5b6eb59b4873
